project/Build.scala

```diff
@@ -4,63 +4,109 @@ import java.io.{ RandomAccessFile, File }
+    buildOpts(YOURKIT_AGENT,
+      List(
+       "-agentlib:jdwp=transport=dt_socket,server=y,suspend=y,address=5005",
+       s"-agentpath:${YOURKIT_AGENT}",
+       "-XX:+HeapDumpOnOutOfMemoryError", "-Xmx1g", "-Xss2m"))
+
+}
+
+object DottyBuild extends Build with CIBuild with DebugBuild {
+
+  lazy val dotty = Project(id = "dotty", base = file("."),
+    settings = defaults)
+
+  def scalaCompilerVersion = "2.11.5-20151022-113908-7fb0e653fd"
+
+  lazy val defaults = Defaults.coreDefaultSettings ++ Seq(
     scalaVersion in Global := "2.11.5",
-    version in Global := "0.1-SNAPSHOT",
+    version in Global      := "0.1-SNAPSHOT",
     organization in Global := "org.scala-lang",
-    organizationName in Global := "LAMP/EPFL",
+    organizationName in Global     := "LAMP/EPFL",
     organizationHomepage in Global := Some(url("http://lamp.epfl.ch")),
-    homepage in Global := Some(url("https://github.com/lampepfl/dotty")),
+    homepage in Global     := Some(url("https://github.com/lampepfl/dotty")),
+
+    // to get Scala 2.11
+    resolvers += Resolver.sonatypeRepo("releases"),
 
     // set sources to src/, tests to test/ and resources to resources/
     scalaSource in Compile := baseDirectory.value / "src",
-    javaSource in Compile := baseDirectory.value / "src",
-    scalaSource in Test := baseDirectory.value / "test",
-    javaSource in Test := baseDirectory.value / "test",
+    javaSource in Compile  := baseDirectory.value / "src",
+    scalaSource in Test    := baseDirectory.value / "test",
+    javaSource in Test     := baseDirectory.value / "test",
     resourceDirectory in Compile := baseDirectory.value / "resources",
-    unmanagedSourceDirectories in Compile := Seq((scalaSource in Compile).value),
-    unmanagedSourceDirectories in Test := Seq((scalaSource in Test).value),
+    unmanagedSourceDirectories in Compile :=
+      Seq((scalaSource in Compile).value),
+    unmanagedSourceDirectories in Test    := Seq((scalaSource in Test).value),
 
     // include sources in eclipse (downloads source code for all dependencies)
-    //http://stackoverflow.com/questions/10472840/how-to-attach-sources-to-sbt-managed-dependencies-in-scala-ide#answer-11683728
-    com.typesafe.sbteclipse.plugin.EclipsePlugin.EclipseKeys.withSource := true,
+    // http://stackoverflow.com/questions/10472840/\
+    // how-to-attach-sources-to-sbt-managed-dependencies\
+    // -in-scala-ide#answer-11683728
+    com.typesafe.sbteclipse.plugin.
+      EclipsePlugin.EclipseKeys.withSource := true,
 
-    // to get Scala 2.11
-    resolvers += Resolver.sonatypeRepo("releases"),
 
     // get libraries onboard
-    partestDeps := Seq("me.d-d" % "scala-compiler" % "2.11.5-20151022-113908-7fb0e653fd",
-                      "org.scala-lang" % "scala-reflect" % scalaVersion.value,
-                      "org.scala-lang" % "scala-library" % scalaVersion.value % "test"),
+    partestDeps :=
+      Seq("me.d-d" % "scala-compiler" % scalaCompilerVersion,
```

Comment:
`bin/dotc` script currenlty detect versions of dependencies by looking for the line containing `scala-compiler` and looking for the last string constant on this line.

Either scalaCompilerVersion needs to be inlined here, or `bin/dotc` script needs to be changed.